src/actions/auth/_util.js

```diff
@@ -0,0 +1,213 @@
+});
+
+//    invalidRefreshType :: InvalidRequestError
+const invalidRefreshType = error(400, {
+  name: 'InvalidRefreshTypeError',
+  message: 'An invalid type of token was provided where a Refresh token was expected'
+});
+
+//    invalidSessionType :: InternalServerError
+const invalidSessionType = error(500, {
+  name: 'InvalidSessionTypeError',
+  message: 'Unexpected session data type'
+});
+
+//    missingAuthorizationHeader :: NotAuthorizedError
+const missingAuthorizationHeader = error(403, {
+  name: 'MissingAuthorizationHeaderError',
+  message: 'Missing Authorization header'
+});
+
+//    missingTokenCookie :: NotAuthorizedError
+const missingTokenCookie = error(403, {
+  name: 'MissingTokenCookieError',
+  message: 'Missing Cookie header'
+});
+
+//    malformedAuthorizationHeader :: InvalidRequestError
+const malformedAuthorizationHeader = error(400, {
+  name: 'MalformedAuthorizationHeaderError',
+  message: 'Malformed Authorization header'
+});
+
+//    malformedAuthorizationHeader :: InvalidRequestError
+const invalidAuthorizationHeader = error(400, {
+  name: 'InvalidAuthorizationHeaderError',
+  message: 'Authorization method must be Bearer'
+});
+
+//    tokenExpired :: NotAuthorizedError
+const tokenExpired = error(403, {
+  name: 'TokenExpiredError',
+  message: 'Token expired'
+});
+
+//    refreshExpired :: NotAuthorizedError
+const refreshExpired = error(403, {
+  name: 'RefreshExpiredError',
+  message: 'Token expired'
+});
+
+//    tokenNotExpired :: NotAuthorizedError
+const tokenNotExpired = error(400, {
+  name: 'TokenNotExpiredError',
+  message: 'Token expired'
+});
+
+//    pairIdMismatch :: NotAuthorizedError
+const pairIdMismatch = error(400, {
+  name: 'PairIdMismatchError',
+  message: 'Token pair identity mismatch'
+});
+
+//    isValidTokenClaims :: Object -> Boolean
+const isValidTokenClaims = pipe([Object.keys, difference(tokenClaimKeys), isEmpty]);
+
+//    isValidRefreshClaims :: Object -> Boolean
+const isValidRefreshClaims = pipe([Object.keys, difference(refreshClaimKeys), isEmpty]);
+
+//    validateTokenClaims :: Either Error Claims -> Either Error Claims
+const validateTokenClaims = chain(claims =>
+  isValidTokenClaims(claims) ? Right(claims) : Left(invalidTokenClaims)
+);
+
+//    validateRefreshClaims :: Either Error Claims -> Either Error Claims
+const validateRefreshClaims = chain(claims =>
+  isValidRefreshClaims(claims) ? Right(claims) : Left(invalidRefreshClaims)
+);
+
+//    join :: Chain m => m (m a) -> m a
+const join = chain(I);
+
+//      createTokenPair :: (b -> Either Error a) -> b -> Future Error [a, a]
+exports.createTokenPair = curry((encode, session) => Future.do(function*() {
+
+  const id = yield randomString(16);
+
+  const refreshClaims = {
+    _: id,
+    t: Refresh,
+    exp: Date.now() + refreshLife
+  };
+
+  const authClaims = {
+    _: id,
+    t: Authorization,
+    $: session,
+    iat: Date.now(),
+    exp: Date.now() + tokenLife
+  };
```

Comment:
Maybe I should add and verify an [`aud`](https://tools.ietf.org/html/rfc7519#section-4.1.3)ience claim.